FreeRTOS_Sockets.c

```diff
@@ -2933,6 +3199,49 @@ uint32_t FreeRTOS_inet_addr( const char * pcIPAddress )
 }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Convert an ASCII character to its corresponding hexadecimal value.
+ *        Accepted characters are 0-9, a-f, and A-F.
+ *
+ * @param[in] cChar: The character to be converted.
+ *
+ * @return The hexadecimal value, between 0 and 15.
+ *         When the character is not valid, socketINVALID_HEX_CHAR will be returned.
+ */
+static uint8_t ucASCIIToHex( char cChar )
+{
+    char cValue = cChar;
+    uint8_t ucNew;
+
+    if( ( cValue >= '0' ) && ( cValue <= '9' ) )
+    {
+        cValue -= ( char ) '0';
+        ucNew = ( uint8_t ) cValue;
+    }
+    else if( ( cValue >= 'a' ) && ( cValue <= 'f' ) )
+    {
+        cValue -= ( char ) 'a';
+        ucNew = ( uint8_t ) cValue;
+        ucNew += ( uint8_t ) 10;
+    }
+    else if( ( cValue >= 'A' ) && ( cValue <= 'F' ) )
+    {
+        cValue -= ( char ) 'A';
+        ucNew = ( uint8_t ) cValue;
+        ucNew += ( uint8_t ) 10;
+    }
+    else
+    {
+        /* The character does not represent a valid hex number, return 255.
+         */
+        ucNew = ( uint8_t ) socketINVALID_HEX_CHAR;
```

Comment:
socketINVALID_HEX_CHAR could be cast to a uint8_t when it is defined.

Reply:
We have this habit now: declare all macros as either signed or unsigned:
~~~c
    #define NO_ERROR    0
    #define MAX_COUNT   1000U
~~~
Later you can decide what actual cast is needed, if any.

And also, I am not sure that after this declaration:
~~~c
#define VALUE ( ( uint8_t ) 255U )
~~~
you can use `VALUE` in an `#ifdef` expression, like e.g. `#if( VALUE > 512U )`.